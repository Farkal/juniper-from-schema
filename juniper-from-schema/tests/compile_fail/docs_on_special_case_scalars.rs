#![allow(dead_code, unused_variables, unused_must_use, unused_imports)]
include!("../setup.rs");

use url::Url;

graphql_schema! {
    schema {
        query: Query
    }

    type Query {
        foo: String! @juniper(ownership: "owned")
    }

    "Url docs"
    scalar Url

    "DateTime docs"
    scalar DateTime

    "Date docs"
    scalar Date

    "Uuid docs"
    scalar Uuid
}

pub struct Query;

impl QueryFields for Query {
    fn field_foo(&self, _: &Executor<'_, Context>) -> FieldResult<String> {
        unimplemented!()
    }
}
