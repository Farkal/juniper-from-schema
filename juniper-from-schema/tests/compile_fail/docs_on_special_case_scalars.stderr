error: proc macro panicked
  --> $DIR/docs_on_special_case_scalars.rs:6:1
   |
6  | / juniper_from_schema::graphql_schema! {
7  | |     schema {
8  | |         query: Query
9  | |     }
...  |
25 | |     scalar Uuid
26 | | }
   | |_^
   |
   = help: message: 
           
           [91merror[0m: Special case scalars don't support having descriptions because the Rust types are defined in external crates
            --> schema:2:63
             |
           2 |    { foo : String ! @ juniper (ownership : "owned") } "Url docs" scalar Url
             |                                                                  [91m^[0m
           
           
           [91merror[0m: Special case scalars don't support having descriptions because the Rust types are defined in external crates
            --> schema:3:20
             |
           3 |    "DateTimeUtc docs" scalar DateTimeUtc "Date docs" scalar Date "Uuid docs"
             |                       [91m^[0m
           
           
           [91merror[0m: Special case scalars don't support having descriptions because the Rust types are defined in external crates
            --> schema:3:51
             |
           3 |    "DateTimeUtc docs" scalar DateTimeUtc "Date docs" scalar Date "Uuid docs"
             |                                                      [91m^[0m
           
           
           [91merror[0m: Special case scalars don't support having descriptions because the Rust types are defined in external crates
            --> schema:4:1
             |
           4 |    scalar Uuid
             |    [91m^[0m
           
           
           aborting due to 4 errors
           

error: could not compile `juniper-from-schema-tests`.
